Linkta-org/core
# Server-side Authentication Support

This pull request implements a middleware function to control access to protected resources on the Linkta server.

On any incoming request which will access the database or the LLM, the request must include an "Authorization" header which carries a valid Firebase idToken. The auth middleware decodes and verifies the idToken.

If verification is successful, the user is considered to be authorized. If the provided token is not a valid Firebase idToken, or if no token is provided, the user is considered to be unauthorized and an error response is returned to the client.

At this time, all authenticated users have equal permissions and access.



Fixes #256 

## Type of change




## How Can this be tested? Testing Plan to review this PR



- [x] Where in the app does this change take place? | Server-side middleware
Send a request to the server lacking a valid Firebase idToken in the Authorization header - expect an error, 401 response
Send a request to the server including a valid Firebase idToken in the Authorization header - expect no error, 200 response


## Checklist:
